sonar-plugin-api/src/main/java/org/sonar/api/batch/sensor/issue/NewIssue.java

```diff
@@ -86,4 +86,11 @@ public interface NewIssue {
    */
   void save();
 
+  /**
+   * Register a specific context for this issue.
+   * e.g. "Spring"
+   * @since 9.6
+   */
+  NewIssue setRuleDescriptionContextKey(String ruleDescriptionContextKey);
```

Comment:
I would make the argument nullable, this allows to not force an extra-if on clients.